### ISSUE TITLE
Lambda Meraki Route Extraction Problem

### ISSUE THREAD

**Issue by NickB118** [aws-quickstart/quickstart-cisco-meraki-sd-wan-vmx#35]
Our existing SD-WAN topology is Hub and Spoke, with the Hub currently set to a pair of physical MX appliances in an on-prem DC and both vMX's set as spokes.

When the below Lambda function ran with our topology, no routes were extracted at all, so no routing updates were made.

I wondered whether this might be to do with what the get_all_vpn_routes function is expecting the topology to be like?

https://github.com/aws-quickstart/quickstart-cisco-meraki-sd-wan-vmx/blob/510a621e2e5455bfb6907c81a0b05284e0bd7c5d/functions/source/lambda_function.py#L57

**Comment by myhomenwlab** [comment]
I have no way of knowing the design philosophy, but reading the code, the following conditions must be met.

* vMX must be Hub.

* Spoke MX on the Branch side specifies vMX as Hub.

In short, only Hub & Spoke with **vMX** as Hub is supported.
In other words, Hub & Spoke with **MX** as Hub is **not** supported.

**Comment by goopilot** [comment]
Yes, we have same issue. And I see that it's not only us https://github.com/aws-quickstart/quickstart-cisco-meraki-sd-wan-vmx/issues/24
